projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -0,0 +1,213 @@
+            overlaySettings.scrollStrategy.attach();
+        }
+
+        this.onOpened.emit();
+        return this._elements[this._elements.length - 1].id;
+    }
+
+    hide(id: string) {
+        // TODO: cleanup
+        this.onClosing.emit();
+
+        const element = this.getElementById(id);
+        if (!element) {
+            console.warn('igxOverlay.hide was called with wrong id: ' + id);
+        }
+
+        const scrollStrategy = element.settings.scrollStrategy;
+        if (scrollStrategy) {
+            scrollStrategy.detach();
+        }
+
+        const child: HTMLElement = element.elementRef.nativeElement;
+        if (!this.OverlayElement.contains(child)) {
+            console.error('Component with id:' + id + 'is already removed!');
+            return;
+        }
+
+        this.OverlayElement.removeChild(child.parentNode.parentNode);
+        this.onClosed.emit();
+    }
+
+    hideAll() {
+        while (this._elements.length > 0) {
+            this.hide(this._elements[this._elements.length - 1].id);
+        }
+    }
+
+    getElementById(id: string) {
+        const element = this._elements.find(e => e.id === id);
+        return element;
+    }
+
+    private getId(id?: string) {
+        return id ? id : (this._componentId++).toString();
+    }
+
+    private getElement(component: any, id: string, overlaySettings: OverlaySettings): HTMLElement {
+        let element: HTMLElement;
+
+        if (this._elements.find(e => e.id === id)) {
+            return this._elements.find(e => e.id === id).elementRef.nativeElement;
+        }
+
+        if (component instanceof ElementRef) {
+            element = component.nativeElement;
+            this._elements.push({
+                id: id,
+                elementRef: <ElementRef>component,
+                componentRef: null,
+                size: element.getBoundingClientRect(),
+                settings: overlaySettings
+            });
+            return element;
+        }
+
+        let dynamicFactory: ComponentFactory<{}>;
+        try {
+            dynamicFactory = this._factoryResolver.resolveComponentFactory(component);
+        } catch (error) {
+            console.error(error);
+            return null;
+        }
+
+        const dc: ComponentRef<{}> = dynamicFactory.create(this._injector);
+
+        element = dc.location.nativeElement;
+        this._appRef.attachView(dc.hostView);
+
+        this._elements.push({
+            id: id,
+            elementRef: dc.location,
+            componentRef: dc,
+            size: element.getBoundingClientRect(),
+            settings: overlaySettings
+        });
+        return element;
+    }
+
+    private getPositionStrategy(positionStrategy?: IPositionStrategy): IPositionStrategy {
+        positionStrategy = positionStrategy ? positionStrategy : new GlobalPositionStrategy();
+        if (positionStrategy._settings && positionStrategy._settings.element) {
```

Comment:
Make this a generic utility static method that takes the settings and returns a point form the target (either directly, or doing these calculations based on element)

Reply:
done